pkg/diff/namespace/diffexpr.go

```diff
@@ -144,12 +149,20 @@ func DiffExpressions(existing *core.UsersetRewrite, updated *core.UsersetRewrite
 		return nil, spiceerrors.MustBugf("unknown operation type %T", updated.RewriteOperation)
 	}
 
+	childChangeKind := ExpressionChildrenChanged
 	if existingType != updatedType {
-		return &ExpressionDiff{
-			existing: existing,
-			updated:  updated,
-			change:   ExpressionOperationChanged,
-		}, nil
+		// If the expression has changed from a union with a single child, then
+		// treat this as a special case, since there wasn't really an operation
+		// before.
+		if existingType != "union" || len(existingOperation.Child) != 1 {
```

Comment:
why not returning expanded with intersections and exclusions too?

Reply:
We don't create intersections/exclusions with a single child. Only union because we needed *some* operation to place there during compilation